snuba/admin/google.py

```diff
@@ -1,49 +1,71 @@
+from abc import ABC
+from typing import Optional
 from urllib.parse import urlencode
 
 import structlog
-from googleapiclient.discovery import Resource
+from googleapiclient.discovery import Resource, build
 
 logger = structlog.get_logger().bind(module=__name__)
 
 
-def get_group_id(service: Resource, group_email: str) -> str:
-    try:
-        query_params = f"groupKey.id={group_email}"
-        request = service.groups().lookup()
-        request.uri += "&" + query_params
-        response = request.execute()
-        if "error" in response:
-            logger.exception(
-                f"An HTTP error occured when fetching group id for email {group_email}.",
-                google_api_error=response["error"],
-            )
+class GoogleAPIWrapper(ABC):
```

Comment:
This should define functions using the `@abstractmethod` decorator. Otherwise this doesn't define any interface.

Reply:
Oh oops, I meant to delete this.